app/src/main/java/com/kodeco/android/countryinfo/ui/screens/countryinfo/CountryInfoViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.kodeco.android.countryinfo.ui.screens.countryinfo
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.launch
+import repositories.CountryRepository
+
+class CountryInfoViewModel(
+    private val repository: CountryRepository
+) : ViewModel() {
+    private val _uiState = MutableStateFlow<CountryInfoState>(CountryInfoState.Loading(0))
+    val uiState: StateFlow<CountryInfoState> = _uiState.asStateFlow()
+
+    private val _appUptimeCounterFlow = MutableStateFlow(0)
+
+    init {
+        startCounterUpdate()
+        fetchCountries()
+    }
+
+    fun refresh() {
+        _uiState.value = CountryInfoState.Loading(_appUptimeCounterFlow.value)
+        fetchCountries()
+//        fetchCountriesWithDelay() // To reviewer: Use this instead of above to see uptime counter :)
```

Comment:
PS: If you're pulling down my branch - you might want to comment in this line (and comment out the one above). This is how I made the recording seen in the PR description.